platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/IsolatedArray.java

```diff
@@ -57,7 +58,7 @@ public Object[] isolate() {
     @Nullable
     @Override
     public <S> S coerce(Class<S> type) {
-        return null;
```

Comment:
💭 I suspect there aren't many List-valued attributes and this omission was never found.  It showed up in tests only we start coercing more aggresively.